docs/resource/app.md

```diff
@@ -0,0 +1,42 @@
+# App
+
+<!-- semconv entity.app -->
```

Comment:
this won't be necessary once https://github.com/open-telemetry/semantic-conventions/pull/2278 is merged, so I'm going to remove entities from the check and delete this file